sstables/sstables.cc

```diff
@@ -2149,18 +2166,14 @@ components_writer::~components_writer() {
     }
 }
 
-future<>
+future<std::optional<sstables::scylla_metadata>>
 sstable::read_scylla_metadata(const io_priority_class& pc) {
-    if (_components->scylla_metadata) {
-        return make_ready_future<>();
+    if (!has_component(component_type::Scylla)) {
+        return make_ready_future<std::optional<sstables::scylla_metadata>>();
     }
-    return read_toc().then([this, &pc] {
-        _components->scylla_metadata.emplace();  // engaged optional means we won't try to re-read this again
-        if (!has_component(component_type::Scylla)) {
-            return make_ready_future<>();
-        }
-        return read_simple<component_type::Scylla>(*_components->scylla_metadata, pc);
-    });
+    sstables::scylla_metadata scylla_metadata;
+    read_simple<component_type::Scylla>(scylla_metadata, pc).get();
+    return make_ready_future<std::optional<sstables::scylla_metadata>>(scylla_metadata);
```

Comment:
This is awkward and will only work when called within an async thread.
How about passing a reference to a `std::optional<sstables::scylla_metadata>` to the function and engage it here and then`return read_simple<component_type::Scylla>(*scylla_metadata, pc);`

Reply:
Isn't github wonderful.

Reply:
I changed this, @bhalevy